__tests__/rpcProvider.test.ts

```diff
@@ -0,0 +1,248 @@
+import { isBN } from 'bn.js';
+
+import { Account, Contract, DeployContractResponse, Provider, RPCProvider } from '../src';
+import { toBN } from '../src/utils/number';
+import { compileCalldata } from '../src/utils/stark';
+import { compiledErc20, describeIfNotDevnet, getTestAccount } from './fixtures';
+
+const { TEST_RPC_URL } = process.env;
+
+if (!TEST_RPC_URL) {
+  throw new Error('TEST_RPC_URL is not set');
+}
+
+describe('RPCProvider', () => {
+  let provider: RPCProvider;
+
+  beforeAll(async () => {
+    provider = new RPCProvider({ nodeUrl: TEST_RPC_URL });
+  });
+
+  describe('Provider methods', () => {
```

Comment:
we should share this test block for both providers, so we just need to write tests once.
It also ensures to a better degree that both work the same on this methods

Reply:
[b525a60](https://github.com/0xs34n/starknet.js/pull/243/commits/b525a60eb77647962f4509bb02f64bb66071c9c2)